test/Behodler.test.js

```diff
@@ -106,68 +157,68 @@ describe('Behodler1', async function () {
-        expect(scarcityBalance).to.equal(expectedScarcity)
+        assertBNequal(scarcityBalance, expectedScarcity)
 
-        const scarcitySupplyAfter = (await bigNum.BNtoBigInt(this.behodler.totalSupply.call())).toString()
-        expect(scarcitySupplyAfter).to.equal(expectedScarcity)
+        const scarcitySupplyAfter = (await bigNum.BNtoBigInt(this.behodler.totalSupply())).toString()
+        assertBNequal(scarcitySupplyAfter, expectedScarcity)
     })
 
     it('adding liquidity as Eth produces correct scarcity', async function () {
-        const weth = await this.behodler.Weth.call()
+        const weth = await this.behodler.Weth()
         expect(weth).to.be.a("string").that.equals(this.weth.address)
 
-        const ethBalanceBefore = await bigNum.BNtoBigInt(balance.current(trader1))
-
-        await this.behodler.addLiquidity(weth, FINNEY * 2n, { from: trader1, value: (FINNEY * 2n).toString() })
+        const ethBalanceBefore = await bigNum.BNtoBigInt(balance.current(trader1.address))
 
-        const ethBalanceAfter = (await bigNum.BNtoBigInt(balance.current(trader1))).toString()
+        const scarcityBalanceBefore = (await this.behodler.balanceOf(trader1.address)).toString()
 
-        const scarcityBalance = (await this.behodler.balanceOf.call(trader1)).toString()
+        await this.behodler.connect(trader1).addLiquidity(weth, FINNEY * 2n, { value: (FINNEY * 2n).toString() })
 
+        const ethBalanceAfter = (await bigNum.BNtoBigInt(balance.current(trader1.address))).toString()
+        const scarcityBalance = (await this.behodler.balanceOf(trader1.address)).toString()
         const expectedScarcity = '201609232779564367355'
-        expect(scarcityBalance).to.equal(expectedScarcity)
+        assertBNequal(scarcityBalance, expectedScarcity)
 
-        const scarcitySupplyAfter = (await bigNum.BNtoBigInt(this.behodler.totalSupply.call())).toString()
-        expect(scarcitySupplyAfter).to.equal(expectedScarcity)
+        const scarcitySupplyAfter = (await bigNum.BNtoBigInt(this.behodler.totalSupply())).toString()
+        assertBNequal(scarcitySupplyAfter, expectedScarcity)
     })
 
     it("withdrawing scarcity transfers out the correct number of tokens", async function () {
         //scarcity supply shrinks
-        await this.regularToken.approve(this.behodler.address, 22n * FINNEY, { from: trader1 })
-        await this.behodler.addLiquidity(this.regularToken.address, 22n * FINNEY, { from: trader1 })
-        const tokenBalanceBeforeWithdraw = await bigNum.BNtoBigInt(this.regularToken.balanceOf(trader1))
-        const scarcityBalanceBeforeWithdraw = await bigNum.BNtoBigInt(this.behodler.balanceOf(trader1))
+        await this.regularToken.connect(trader1).approve(this.behodler.address, 22n * FINNEY)
+        await this.behodler.connect(trader1).addLiquidity(this.regularToken.address, 22n * FINNEY)
+        const tokenBalanceBeforeWithdraw = await bigNum.BNtoBigInt(this.regularToken.balanceOf(trader1.address))
```

Comment:
bigNum.BNtoBigIn

Reply:
some variables/consts are using in calculation, so above statement was removed where it is possible not fail the tests